fastn/django/__init__.py

```diff
@@ -110,13 +109,12 @@ def _add_user(self, request: RequestType):
         )
 
         user, _ = User.objects.get_or_create(
-            username=fastn_user.get("login"),
+            username=fastn_user.get("username"),
             defaults={
                 "first_name": first_name,
                 "last_name": last_name,
-                "email": fastn_user.get("email") or "",
-                # email has Not Null constraint
-            }
+                "email": fastn_user.get("email"),
+            },
         )
```

Comment:
Let's convert this to a hook, 

```py
import django.utils
from django.conf import settings

django.utils.module_loading(
    settings.get_attr(
        "FASTN_AUTH_CALLBACK", 
        "fastn.django.default_callback"
    )
)(request=request, fastn_user=fastn_user)
```

Where `fastn.django.default_callback` will be some code like this:

```py
def default_callback(request, fastn_user):
       user, _ = User.objects.get_or_create(
            username=fastn_user.get("login"),
            username=fastn_user.get("username"),
            defaults={
                "first_name": first_name,
                "last_name": last_name,
                "email": fastn_user.get("email") or "",
                # email has Not Null constraint
            }
                "email": fastn_user.get("email"),
            },
        )
```

Reply:
should I also add call to `django.contrib.auth.login` in `default_callback`?

Reply:
Not sure. I think we will always need `auth.login()`.

Reply:
What does it do?

Reply:
if we choose to move `auth.login` inside the callback. Our middleware will just be responsible for decoding the fastn set cookie into a python dict and then the callback can decide to do whatever it likes with it which I think is a better way.

Reply:
we can also get rid of the utility function `utils.get_first_name_and_last_name`

Reply:
Let's do this.

Reply:
done :)